OpenRA.Mods.Common/Traits/BotModules/BuildingRepairBotModule.cs

```diff
@@ -26,6 +26,12 @@ public BuildingRepairBotModule(Actor self, BuildingRepairBotModuleInfo info)
 
 		void IBotRespondToAttack.RespondToAttack(IBot bot, Actor self, AttackInfo e)
 		{
+			// HACK: We don't want D2k bots to repair all their buildings on placement
+			// where half their HP is removed via neutral terrain damage.
+			// TODO: Implement concrete placement for D2k bots and remove this hack.
+			if (e.Attacker.Owner.Stances[self.Owner] != Stance.Enemy)
```

Comment:
The original `HackyAI` check used `if (e.Attacker.Owner.Stances[self.Owner] == Stance.Neutral)`. Wouldn't it be safer to keep with this?

Reply:
Hm, I guess, I just misremembered.